### DISCUSSION TITLE
AIR shared runtimes

### DISCUSSION THREAD

**Discussion by ajwfrost** [airsdk/Adobe-Runtime-Support#3156] [General]
Hi all

This is a thread that will provide information about the AIR shared runtimes, i.e. the standalone desktop installer applications that allow you to then install .air files as if they were native apps.

The current status (as over 4th April 2024) is that we have two variants, available at https://airsdk.harman.com/runtime

1. Version 50.2.4.1. This is (currently) the latest available version. There is an issue with the macOS version here though, where it is only including the x86_64 variant of the app, rather than being a universal binary.

2. Version 33.1.1.744. This is an old version, provided solely for applications that still need the old embedded 'webkit' version for display of web pages and interactivity between ActionScript and JavaScript.

It is our hope that most AIR application developers will package their apps with the captive runtime available using the AIR SDK build tools, to remove the dependency upon this shared runtime. However, we are looking to provide the universal binary version for macOS and to keep updating these runtimes from time to time with updates from the AIR SDK releases. There is also the possibility of creating a version for Linux as well, if that's something that would be useful.

If you want notification of new releases to the shared runtime, you can subscribe to this thread for notifications. Adobe had had an auto-update mechanism in place, but disabled that just before handing the software over to us - and of course for us to adopt this, we would need to adapt it and update/implement the server-side capabilities. But this is something that could be done if there was sufficient demand for it..

thanks

**Comment by Fancy2209** [comment]
Wouldn't a Linux Shared Runtime allow to bypass the need for a license?

**Reply by ajwfrost** [reply]
No, I would expect this to work the same as currently i.e. the AIR application developer would have needed to package the app with a license.

Currently on Windows/macOS, if you take an old .air file from 32.0 or earlier - or if you're "free tier" with 33.1+ - then the app will run with a splash screen. If you're a commercial licensee of ours and create a .air file, it will run without this.

On Linux, instead of displaying the splash screen, we just block the runtime.... so that could stay, meaning that the shared runtime is really just for the benefit of our paying licensees :-)

thanks

**Comment by ajwfrost** [comment]
AIR shared runtime update: release 51.0.1.2 is now available at https://airsdk.harman.com/runtime
For release notes, see the updates from the AIR SDK.

On macOS, this is provided solely as an x86_64 installer. The runtime framework within it has arm64 support but that is unlikely to be used: all apps created via the installation of a .air file should end up as x86_64 and will need to run using Rosetta2 on Apple Silicon computers. The reason for this is that the arm64 based binaries are blocked at start-up by the Gatekeeper software, because they don't have a code signature. We still recommend using the AIR SDK to create signed application bundles and deploying these yourself using PKG files or DMG files etc, so that you can correctly sign/notarize the packages.

thanks

**Comment by ajwfrost** [comment]
Hi all - we've just updated the website to publish the shared runtimes for 51.1.2.2: https://airsdk.harman.com/runtime
Any issues, please let us know..

**Comment by ajwfrost** [comment]
We've just released the 51.2.1.1 version of AIR SDK. For any apps that are built as .air files using the 51.2 namespace, you'll need to have an updated shared AIR runtime, so we've also posted the equivalent installers for Windows/macOS at https://airsdk.harman.com/runtime

**Reply by AnAmateurDev** [reply]
I am unable to open an old air app ? was anything changed for `URLLoader` ?

**Reply by ajwfrost** [reply]
Sorry @AnAmateurDev, hadn't noticed this post before! If it's an old app that doesn't work with the newer runtime, it is likely to have been using the HTML-based features provided by a WebKit library that used to be part of AIR. You can download version33.1.1.744 of the AIR runtime if you scroll down on the runtimes page, which should then support it?
thanks

**Comment by ajwfrost** [comment]
Hi all - 51.2.2.6 has been released as a runtime update as well. If you want to install this, please check the https://airsdk.harman.com/runtime site.
thanks

**Comment by ajwfrost** [comment]
Apologies, we missed to notify of the first 51.3.1 runtime that had been uploaded - but we have now got 51.3.2.2 available online.

We have since found one issue with it, regarding the macOS application installer/uninstaller: if you try to open an .air file, then normally it would either (a) install the app, if it's not installed, or (b) give you options to launch or uninstall the app, if it's already installed. With 51.3.2.2, this second workflow does not work - it will report an error and will mean you cannot uninstall the AIR application using this method. We will issue an update to the shared runtime shortly, and once that is installed, the .air file handling behaviour will continue as it should.

thanks

**Reply by ajwfrost** [reply]
FYI the issue with 51.3.2.2 mentioned above also caused a problem when installing using a 'native' installer (i.e. a .exe file on Windows) so please update to 51.3.2.3 which we published earlier today, which fixes these problems.

thanks

**Comment by dmunsie** [comment]
I've only built Android apps with the Air SDK. Is there a tutorial somewhere how to actually build a PC version using the Air SDK? Thanks for the continued SDK development and support.

**Comment by ajwfrost** [comment]
Hi all
AIR runtime 51.3.3.1 is available now. This would help for folk who are running on macOS with apple silicon and who are facing issues with warnings about Rosetta (although it may need the app developers to also re-package their .air files ... please see the [release notes](https://airsdk.harman.com/release_notes?q=51.3.3.1)!)
